### PR TITLE
Install YARN on cgcloud-jenkins/toil_jenkins_slave (resolves #149, resolves #161)

### LINKED ISSUE #149: Add YARN installation to cgcloud-jenkins
This is needed for testing https://github.com/BD2KGenomics/toil/issues/796.

### LINKED ISSUE #161: Install Spark on cgcloud-jenkins for Toil
This is necessary for testing https://github.com/BD2KGenomics/toil/pull/786, which resolves https://github.com/BD2KGenomics/toil/issues/774. I imagine that there's logic in `cgcloud-spark` that I'll be cribbing from.

### PR DESCRIPTION
**Pull request by hannes-ucsc**
Continued from #150
Resolves #149 and #161

### CODE REVIEW

**Comment by hannes-ucsc** [comment]
Jenkins, test this please.

**Comment by fnothaft** [comment]
Thanks for taking this, @hannes-ucsc!

File: core/src/cgcloud/core/apache.py [hunk 1]
```diff
@@ -47,15 +52,14 @@ def __install_apache_package( self, path ):
...
+        mirrors = run( "curl -fs '%s'" % url )
         mirrors = json.loads( mirrors )
         mirror = mirrors[ 'preferred' ]
-        url = mirror + path
+        url = mirror + remote_path
         return url
 
-    def __apache_s3_mirror_url( self, path ):
-        file_name = os.path.basename( path )
+    def _apache_s3_mirror_url( self, remote_path ):
```
Comment (fnothaft) [inline core/src/cgcloud/core/apache.py]: In spite of my stupidity RE: protected vs. private in Python, I do think that this method should be private.

**Comment by hannes-ucsc** [comment]
Jenkins, test this please.

File: jenkins/src/cgcloud/jenkins/toil_jenkins_slave.py [hunk 2]
```diff
@@ -84,26 +86,32 @@ def __install_parasol( self ):
...
-            spark_path = [ fmt( '{install_dir}/spark' ) ]
-            pyspark_path = [ fmt( '{spark_path}/python' ) ]
-            self._patch_etc_environment( f, env_pairs={ 'SPARK_HOME': spark_path } )
-            self._patch_etc_environment( f, pyspark_path, envar='PYTHONPATH' )
+            spark_home = fmt( '{install_dir}/spark' )
+            # These two PYTHONPATH entries are also added by the 'pyspark' wrapper script.
+            # We need to replicate them globally because we want to be able to just do 
+            # 'import pyspark' in Toil's Spark service code and associated tests.
+            python_path = [ fmt( '{spark_home}/python' ),
+                            run( fmt( 'ls {spark_home}/python/lib/py4j-*-src.zip' ).strip() ) ]
```
Comment (fnothaft) [inline jenkins/src/cgcloud/jenkins/toil_jenkins_slave.py:110]: Ah! Thanks for picking up on this. I didn't realize that the `py4j*.zip` could be added to the PYTHONPATH.

**Comment by fnothaft** [comment]
8f473db LGTM. Thanks for cleaning this up, @hannes-ucsc!

**Comment by hannes-ucsc** [comment]
Jenkins, test this please.

File: jenkins/src/cgcloud/jenkins/toil_jenkins_slave.py (hunk 2 above)
Comment (hannes-ucsc) [inline jenkins/src/cgcloud/jenkins/toil_jenkins_slave.py:110]: Thank you for noticing!

File: core/src/cgcloud/core/apache.py (hunk 1 above)
Comment (hannes-ucsc) [inline core/src/cgcloud/core/apache.py]: Agreed.

File: jenkins/src/cgcloud/jenkins/toil_jenkins_slave.py (hunk 2 above)
Comment (fnothaft) [inline jenkins/src/cgcloud/jenkins/toil_jenkins_slave.py:110]: I try to learn a bit, give credit where credit is due, etc. Thanks for taking this the last mile!

**Comment by hannes-ucsc** [comment]
Jenkins, test this please.

**Comment by hannes-ucsc** [comment]
Jenkins, test this please.

**Comment by hannes-ucsc** [comment]
Jenkins, test this please.

**Pull request merged**